airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/migrate_to_poetry/pipeline.py

```diff
@@ -0,0 +1,475 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+from __future__ import annotations
+
+import re
+from typing import TYPE_CHECKING
+
+import dagger
+import git
+import requests
+import toml
+from connector_ops.utils import ConnectorLanguage  # type: ignore
+from jinja2 import Environment, PackageLoader, select_autoescape
+from pipelines.airbyte_ci.connectors.build_image.steps.python_connectors import BuildConnectorImages
+from pipelines.airbyte_ci.connectors.bump_version.pipeline import AddChangelogEntry, BumpDockerImageTagInMetadata, get_bumped_version
+from pipelines.airbyte_ci.connectors.consts import CONNECTOR_TEST_STEP_ID
+from pipelines.airbyte_ci.connectors.context import ConnectorContext, PipelineContext
+from pipelines.airbyte_ci.connectors.reports import ConnectorReport, Report
+from pipelines.consts import LOCAL_BUILD_PLATFORM
+from pipelines.dagger.actions.python.common import with_python_connector_installed
+from pipelines.helpers.execution.run_steps import StepToRun, run_steps
+from pipelines.models.steps import Step, StepResult, StepStatus
+
+if TYPE_CHECKING:
+    from typing import Iterable, List, Optional
+
+    from anyio import Semaphore
+
+PACKAGE_NAME_PATTERN = r"^([a-zA-Z0-9_.\-]+)(?:\[(.*?)\])?([=~><!]=?[a-zA-Z0-9\.]+)?$"
+
+
+class CheckIsMigrationCandidate(Step):
+
+    context: ConnectorContext
+
+    title = "Check if the connector is a candidate for migration to poetry."
```

Comment:
is there a high level description of what it takes for a connector to be a candidate? Can we either point to it from this file or add a comment?

Reply:
I'll add a comment 👍